### ISSUE TITLE
Improve error messaging for common OLS mistake w/ categorical data. 

### ISSUE THREAD

**Issue by paultopia** [statsmodels/statsmodels#6690]
#### Is your feature request related to a problem? Please describe

A common mistake of learners or just people who are working fast and not paying a lot of attention is to try to use OLS regression with a categorical dependent variable, often described by a string.  The error messaging for this can be really unhelpful though.  

Toy example from a problem that a student of mine just got bitten by, where `df` is a pandas data frame in wide format with 6345 observations, `foo` is a categorical variable with 3 classes denoted by strings, and `bar` is a float: 

```
import statsmodels.formula.api as sm
reg = sm.ols(formula='foo ~ bar', data=df)
f = reg.fit()
print(f.summary())
```

This is a mistake, the student should be using `mnlogit` rather than `ols`.  It should be a pretty readily detectable mistake from the statsmodels side: it's never right to have a column of strings as the dependent variable in an OLS regression.  But the error they get is really painful: 

```
ValueError: shapes (6345,3) and (6345,3) not aligned: 3 (dim 1) != 6345 (dim 0)
```

This is obviously some kind of error thrown from deep within numpy that is not only meaningless, but even actively confusing to people who aren't experienced with all the ways that numpy represents data shapes, broadcasting, etc. etc. (natural first thought: "how could (6345,3) and (6345,3) not be aligned?  they're the same tuple!").

Making matters worse, they don't get this error on the creation of the model, or on the call to `fit()` where it might make sense---as far as they know, they successfully fit a model, at least up to the point they try to actually view the output with `summary()` and only then is an error thrown.

(Prior issue where someone ran into this: #4794 )

#### Describe the solution you'd like

Statsmodels should check the type of the left-hand variable when an OLS function is called, and if it isn't numeric (or I guess boolean if people really really insist on using OLS instead of logit or whev for two-class categoricals) it should throw an error right away, either on initializing the model with `ols()` or on fitting it with `fit()`.  And that error should be geared toward people who might not realize their data isn't numeric, or who might not realize that it's a bad idea to throw OLS at everything.  For example, it could be a `ValueError` with a message like "Dependent variable in OLS regression must be numeric." 

#### Describe alternatives you have considered

If there are conceivable use cases where someone might want to try to run an OLS with a column of strings on the left side (and can somehow get them to work?!), I guess throw a warning?  But really, this should just be a clear error.

#### Additional context

I'm happy to work on a PR myself if there's interest in this.

**Comment by bashtage** [comment]
This would need to go into `OLS` as a check that `endog` was either 1-d squeezable to 1-d. There are many models that only work for 1-d input that would benefit from a user-friendly message informing them that their input has the wrong dimension.

**Comment by bashtage** [comment]
This is already fixed in master.

Running the code above produces:

```
ValueError: endog has evaluated to an array with multiple columns that has shape (500, 8). This occurs when the variable converted to endog is non-numeric (e.g., bool or str).
```

**Comment by bashtage** [comment]
closed by 30a9d2992ebe2df1dd3b622111e7fb1249717f4f
